### PR TITLE
Clean up the examples

### PR DESCRIPTION
**Pull request by tcharding**
In an effort to make the examples more digestible to newcomers do various clean ups and refactorings. Each example is done as a separate patch to assist review.

The diffs get smaller as you go down the list :)

### CODE REVIEW

**Review by apoelstra — APPROVED** [review]
ACK f0501dae5fa617900e5f9bc6ae6b60c55d6be2bb

File: examples/verify_tx.rs [hunk 1]
```diff
@@ -12,18 +12,119 @@
...
+    // as having participated in the script
+
+    println!("\n\nExample two:\n");
+    let secp = Secp256k1::new();
+
+    // Sometimes it is necessary to have additional information to get the
+    // `bitcoin::PublicKey` from the `MiniscriptKey` which can be supplied by
+    // the `to_pk_ctx` parameter. For example, when calculating the script
+    // pubkey of a descriptor with xpubs, the secp context and child information
+    // maybe required.
```
Comment (sanket1729) [inline examples/verify_tx.rs:87]: Ouch, we no longer have to_pk_ctx in this codebase. This comment should have been removed. Not a part of this PR.

File: examples/htlc.rs [hunk 2]
```diff
@@ -39,31 +39,34 @@ fn main() {
...
     .expect("Resource limits");
 
-    // Check whether the descriptor is safe
-    // This checks whether all spend paths are accessible in bitcoin network.
-    // It maybe possible that some of the spend require more than 100 elements in Wsh scripts
-    // Or they contain a combination of timelock and heightlock.
+    // Check whether the descriptor is safe. This checks whether all spend paths are accessible in
+    // the Bitcoin network. It may be possible that some of the spend paths require more than 100
+    // elements in Wsh scripts or they contain a combination of timelock and heightlock.
     assert!(htlc_descriptor.sanity_check().is_ok());
```
Comment (sanket1729) [inline examples/htlc.rs:45]: I would really like to get rid of these sanity checks. I think we lose of using rust if users have to call sanity check to double-check things. This feels like CPubKey.isVliad() type stuff

**Review by sanket1729 — APPROVED** [review]
ACK f0501dae5fa617900e5f9bc6ae6b60c55d6be2bb

File: examples/htlc.rs (hunk 2 above)
Comment (apoelstra) [reply examples/htlc.rs:45]: Yeah, I think they don't belong in the example. They'd be useful for users who are manually constructing the `Descriptor` object I think.

Comment (tcharding) [reply examples/htlc.rs:45]: Should we be doing the sanity checks in all the constructors (incl. `FromStr`) instead?

**Pull request merged**